### PR TITLE
Enable graceful shutdown endpoint

### PR DESCRIPTION
**Pull request by masayag**
The PR enables the shutdown endpoint which can be triggered by:
(for `local` profile)
```
curl -u test:test -X POST  localhost:8080/actuator/shutdown
```

In addition, a script for local invocation of the `workflow-service` is added. If the service dies, the script will relaunch it. It may be useful for testing/development locally when there is a need to replace only the workflows jar.

The graceful shutdown timeout (`spring.lifecycle.timeout-per-shutdown-phase`) is set to 25s, since the default timeout for K8s pods is 30s. This can be revisited and tuned by changing the `workflow-service` deployment to specify a different timeout by terminationGracePeriodSeconds.

The graceful shutdown sequence should look as:
```
2023-03-27 22:54:55.497  INFO 861119 --- [       Thread-7] o.s.b.w.e.tomcat.GracefulShutdown        : Commencing graceful shutdown. Waiting for active requests to complete
2023-03-27 22:54:55.507  INFO 861119 --- [tomcat-shutdown] o.s.b.w.e.tomcat.GracefulShutdown        : Graceful shutdown complete
2023-03-27 22:54:55.571  INFO 861119 --- [       Thread-7] c.r.p.w.e.service.WorkFlowServiceImpl    :  >>>>>>>>> Shutting down the workflow service <<<<<<<<<<<
2023-03-27 22:54:55.571  INFO 861119 --- [       Thread-7] c.r.p.w.r.BeanWorkFlowRegistryImpl       :  >>>>>>>>> Shutting down the bean registry <<<<<<<<<<<
2023-03-27 22:54:55.572  INFO 861119 --- [       Thread-7] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2023-03-27 22:54:55.574  INFO 861119 --- [       Thread-7] com.zaxxer.hikari.HikariDataSource       : ${DATASOURCE_NAME} - Shutdown initiated...
2023-03-27 22:54:55.578  INFO 861119 --- [       Thread-7] com.zaxxer.hikari.HikariDataSource       : ${DATASOURCE_NAME} - Shutdown completed.
```

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java [hunk 1]
```diff
@@ -359,4 +360,9 @@ private String validateWorkflow(String workflowName, WorkFlow workFlow) {
 		return null;
 	}
 
+	@PreDestroy
+	public void gracefulShutdown() {
+		log.info(" >>>>>>>>> Shutting down the workflow service <<<<<<<<<<<");
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java:365]: @lshannon @RichardW98 @anludke which logic would we like to have in this context?
Right before the application ends, there are no active requests, but there may be running executions.

Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java:365]: yes we need to wait for running tasks and their persistence to complete before shutdown, and prevent upcoming workflow/task runs

**Comment by masayag** [comment]
> Do you think it makes sense to add shutdown endpoint to notification service?

Could be. Or at least enable a graceful shutdown to it.
I suggest tracking it separately since it wasn't evaluated as part of the workflow-service.

**Pull request merged**